### PR TITLE
Add more tests to react-ui components

### PR DESCRIPTION
**Pull request by m4manjesh**
Closes [TD-1463
](https://sce.myjetbrains.com/youtrack/issue/TD-1463/Add-more-unit-testing-for-App-Layout-and-AppHeader-compnent)

Added more unit tests for the AppLayout and AppHeader components.

- Added tests for user click event props to test it is called
- Added tests for app name and user names to check it displayed properly
- Added tests to check contents are displayed

Package.json updated while resolving some conflicts.

### CODE REVIEW

File: src/AppLayout/AppLayout.test.js [hunk 1]
```diff
@@ -34,8 +35,41 @@ const defaultInputs = {
...
+    expect(onChangePassword).toHaveBeenCalled();
+  });
+  test("onLogout called on user click", async () => {
+    const user = userEvent.setup();
+    const onLogout = jest.fn();
+    render(<AppLayout {...defaultInputs} onLogout={onLogout} />);
+    await user.click(screen.getByRole("button", { name: /JB/i }));
+    await user.click(screen.getByRole("menuitem", { name: /Logout/i }));
+    expect(onLogout).toHaveBeenCalled();
   });
```
[the remark this answers is not in the document]
Comment (m4manjesh) [reply src/AppLayout/AppLayout.test.js:112]: This test is for open the sidebar menu. onAppClick is showing the launcher and it is handled inside the component.

**Comment by mattcorner** [comment]
Does this PR now include updating all dependencies? I note that the package.json has been updated in the latest commit, but not sure why.

**Comment by m4manjesh** [comment]
> Does this PR now include updating all dependencies? I note that the package.json has been updated in the latest commit, but not sure why because it is not related to this PR, nor is it included in the original description.

It got updated while resolving a conflict. PR description updated.

**Comment by m4manjesh** [comment]
> Closes [TD-1463 ](https://sce.myjetbrains.com/youtrack/issue/TD-1463/Add-more-unit-testing-for-App-Layout-and-AppHeader-compnent)
> 
> Added more unit tests for the AppLayout and AppHeader components.
> 
> * Added tests for user click event props to test it is called
> * Added tests for app name and user names to check it displayed properly
> * Added tests to check contents are displayed
> 
> Package.json updated for some dependency updates while resolving conflicts

**Comment by m4manjesh** [comment]
> Does this PR now include updating all dependencies? I note that the package.json has been updated in the latest commit, but not sure why because it is not related to this PR, nor is it included in the original description.

It was a conflict. PR description updated.

**Comment by mattcorner** [comment]
> > Does this PR now include updating all dependencies? I note that the package.json has been updated in the latest commit, but not sure why because it is not related to this PR, nor is it included in the original description.
> 
> It was a conflict. PR description updated.

It was only a conflict because you'd updated all dependencies in 3a233c975c03dbc72b38b0ffa9b015521e8445a4, whilst the DRI @nikitharani was also doing dependency updates. Please avoid doing unrelated work in PRs, it makes reviewing and traceability harder, and you potentially block your own work from being merged because you pull in unknown issues.

**Comment by m4manjesh** [comment]
> > > Does this PR now include updating all dependencies? I note that the package.json has been updated in the latest commit, but not sure why because it is not related to this PR, nor is it included in the original description.
> > 
> > 
> > It was a conflict. PR description updated.
> 
> It was only a conflict because you'd updated all dependencies in [3a233c9](https://github.com/IPG-Automotive-UK/react-ui/commit/3a233c975c03dbc72b38b0ffa9b015521e8445a4), whilst the DRI @nikitharani was also doing dependency updates. Please avoid doing unrelated work in PRs, it makes reviewing and traceability harder, and you potentially block your own work from being merged because you pull in unknown issues.

I was just looking at how this got updated in my code. While showing @nikitharani the way to update all dependencies together i accidentally committed the package-lock.json. Then before submitting to re-review i pulled the latest from Mian and i see the conflict, I thought something got updated in the main. It's all my mistake.

File: src/AppHeader/AppHeader.test.js [hunk 2]
```diff
@@ -14,8 +16,57 @@ const defaultInputs = {
 
 // test app name in the component
```
[the remark this answers is not in the document]
Comment (nikitharani) [reply src/AppHeader/AppHeader.test.js:17]: The comment is not updated. As I mentioned it is not appropriate.

[the remark this answers is not in the document]
Comment (m4manjesh) [reply src/AppHeader/AppHeader.test.js]: done

**Comment by nikitharani** [comment]
LGTM. Merge can be performed.

**Pull request merged**